pkg/build/api/validation/validation.go

```diff
@@ -36,22 +36,43 @@ func ValidateBuildUpdate(build *buildapi.Build, older *buildapi.Build) fielderro
 	return allErrs
 }
 
+func streamTagKey(ref *kapi.ObjectReference) string {
+	if ref.Namespace != "" {
+		return fmt.Sprintf("%s/%s", ref.Namespace, ref.Name)
+	}
+	return ref.Name
+}
+
 // ValidateBuildConfig tests required fields for a Build.
 func ValidateBuildConfig(config *buildapi.BuildConfig) fielderrors.ValidationErrorList {
 	allErrs := fielderrors.ValidationErrorList{}
 	allErrs = append(allErrs, validation.ValidateObjectMeta(&config.ObjectMeta, true, validation.NameIsDNSSubdomain).Prefix("metadata")...)
 
-	// allow only one ImageChangeTrigger for now
+	// allow only one ImageChangeTrigger without a From reference
 	ictCount := 0
+	fromRefs := map[string]struct{}{}
 	for i, trg := range config.Spec.Triggers {
 		allErrs = append(allErrs, validateTrigger(&trg).PrefixIndex(i).Prefix("triggers")...)
-		if trg.Type == buildapi.ImageChangeBuildTriggerType {
-			if ictCount++; ictCount > 1 {
-				allErrs = append(allErrs, fielderrors.NewFieldInvalid("triggers", config.Spec.Triggers, "only one ImageChange trigger is allowed"))
-				break
+		if trg.Type != buildapi.ImageChangeBuildTriggerType {
+			continue
+		}
+		if trg.ImageChange != nil && trg.ImageChange.From != nil {
+			fromKey := streamTagKey(trg.ImageChange.From)
+			_, exists := fromRefs[fromKey]
+			if exists {
+				allErrs = append(allErrs, fielderrors.NewFieldInvalid("triggers", config.Spec.Triggers, "multiple ImageChange triggers refer to the same image stream tag"))
 			}
+			fromRefs[fromKey] = struct{}{}
+			continue
+		}
+
+		// If a From is not specified, we only allow one trigger
+		if ictCount++; ictCount > 1 {
```

Comment:
i think you could do this logic in the loop above.  if trg.ImageChange.From==nil, then use config.Spec.Strategy.From (there is a method to get the From ref from a buildconfig for you, it'll sort out which strategy to look at).

if you do that, then having 2 triggers with empty from's would just be like any other duplicate.